peps/pep-0817.rst

```diff
@@ -0,0 +1,2528 @@
+it:
+
+- If both the build tag and the variant label are present, the filename
+  contains too many components.
+
+  Example:
+  ``numpy-2.3.2-1-cp313-cp313t-musllinux_1_2_x86_64-x86_64_v3.whl``
+
+- If only the variant label is present, the Python tag at third position
+  will be misinterpreted as a build number. Since the build number must
+  start with a digit and no Python tags at the time start with digits,
+  the filename is considered invalid.
+
+  Example:
+  ``numpy-2.3.2-cp313-cp313t-musllinux_1_2_x86_64-x86_64_v3.whl``
+
+This behavior was confirmed for a number of existing tools:
+`auditwheel
+<https://github.com/pypa/auditwheel/blob/6839107e9b918e035ab2df4927a25a5f81f1b8b6/src/auditwheel/repair.py#L61-L64>`__,
+`packaging
+<https://github.com/pypa/packaging/blob/78c2a5e4f5c04fd782a5729d93892c3a3eafe365/src/packaging/utils.py#L94-L134>`__,
+`pdm
+<https://github.com/pdm-project/pdm/blob/66c86908c9b9d07ad8d101d07879d69a55de5c54/src/pdm/models/requirements.py#L259-L286>`__,
+`pip
+<https://github.com/pypa/pip/blob/c46141c29c3646a3328bc4e51d354cc732fb1432/src/pip/_internal/models/wheel.py#L38-L46>`__,
+`poetry
+<https://github.com/python-poetry/poetry/blob/1c04c65149776ae4993fa508bef53373f45c66eb/src/poetry/utils/wheel.py#L23-L27>`__,
+`uv
+<https://github.com/astral-sh/uv/blob/f6a9b55eb73be4f1fb9831362a192cdd8312ab96/crates/uv-distribution-filename/src/wheel.rs#L182-L299>`__.
+
+
+Variant properties system
+-------------------------
+
+Variant properties serve the purpose of expressing the characteristics
+of the variant. Unlike platform compatibility tags, they are stored in
+the variant metadata and therefore do not affect the wheel filename
+length. They follow a hierarchical key-value design, with the key
+further broken into a namespace and a feature name. Namespaces are used
+to group features defined by a single provider, and to avoid conflicts
+should multiple providers define a feature with the same name. This
+permits independent governance and evolution of every namespace.
+
+The keys are restricted to lowercase letters, digits and underscores.
+Uppercase characters are disallowed to avoid different spellings of the
+same name. The character set for values is more relaxed, to permit
+values resembling versions.
+
+Variant features can be declared as allowing multiple values to be
+present within a single variant wheel. If that is the case, these values
+are matched as a logical disjunction, i.e. only a single value needs to
+be compatible with the system for the wheel to be considered supported.
+On the other hand, features are treated conjunctively, i.e. all of them
+need to be compatible. This provides some flexibility in designating
+variant compatibility while avoiding having to implement a complete
+boolean logic.
+
+Variant properties are serialized into a structured 3-tuple format
+inspired by Trove Classifiers in :pep:`301`:
+
+::
+
+    {namespace} :: {feature_name} :: {feature_value}
+
+
+Null variant
+------------
```

Comment:
This section could do with a better overview / motivation. I think 'null' as the name is also confusing, as (implicitly?) there are min requirements for the 'base case'/fallback wheel. I'm also not sure this section flows best here ordering-wise